BitwardenShared/UI/Vault/Vault/VaultItemSelection/VaultItemSelectionView.swift

```diff
@@ -104,25 +104,59 @@ private struct VaultItemSelectionSearchableView: View {
                 get: \.toast,
                 send: VaultItemSelectionAction.toastShown
             ))
+            .background(Color(asset: Asset.Colors.backgroundSecondary).ignoresSafeArea())
     }
 
     // MARK: Private Views
 
     /// The content displayed in the view.
     @ViewBuilder
     private func contentView() -> some View {
-        VStack(spacing: 0) {
-            if isSearching {
-                searchContentView()
-            } else {
-                if store.state.vaultListSections.isEmpty {
-                    // TODO: BIT-2350 Add empty view
-                } else {
-                    matchingItemsView()
+        if isSearching {
+            searchContentView()
+        } else {
+            if store.state.vaultListSections.isEmpty {
+                GeometryReader { reader in
+                    VStack(spacing: 24) {
+                        Asset.Images.openSource.swiftUIImage
+                            .resizable()
+                            .frame(width: 100, height: 100)
+                            .padding(.bottom, 8)
+
+                        Text(
+                            Localizations.thereAreNoItemsInYourVaultThatMatchX(
+                                store.state.ciphersMatchingName ?? "--"
+                            ) +
+                                "\n" +
+                                Localizations.searchForAnItemOrAddANewItem
+                        )
+                        .styleGuide(.callout)
+                        .foregroundColor(Asset.Colors.textPrimary.swiftUIColor)
+                        .multilineTextAlignment(.center)
+
+                        Button {
+                            store.send(.addTapped)
+                        } label: {
+                            HStack(spacing: 8) {
+                                Asset.Images.plus.swiftUIImage
+                                    .imageStyle(.accessoryIcon(
+                                        color: Asset.Colors.textPrimaryInverted.swiftUIColor,
+                                        scaleWithFont: true
+                                    ))
+
+                                Text(Localizations.addAnItem)
+                            }
+                        }
+                        .buttonStyle(.primary(shouldFillWidth: false))
+                    }
+                    .padding(.vertical, 16)
+                    .frame(maxWidth: .infinity, minHeight: reader.size.height)
+                    .scrollView(addVerticalPadding: false)
```

Comment:
🎨 Can this be moved to a separate view/component? This could be reused in different empty lists.

Reply:
Yep, added.